qiskit/aqua/algorithms/minimum_eigen_solvers/iqpe.py

```diff
@@ -193,7 +193,7 @@ def construct_circuit(self,
         qc += self._state_in.construct_circuit('circuit', q)
         # hadamard on a[0]
         qc.add_register(a)
-        qc.u2(0, np.pi, a[0])
+        qc.u(np.pi/2, 0, np.pi, a[0])
```

Comment:
This is just a Hadamard gate, I think it would be better readable if `U2(0, pi)` would be replaced by `H` and similarly `U3(pi, 0, pi)` by `X`. This will be less cryptic for users to understand 😄

Reply:
Code was done solely with U1 etc in the early days, with a view to performance, so as the transpiler could be skipped. I know sometimes there were comments left with X or H. Grover and (I)QPE are being redone anyway so things can be altered there during a refactor as felt best. This PR was really to avoid deprecation rather than say improve cct readability.